docs-archived/guides/dsm_sobject.md

```diff
@@ -0,0 +1,665 @@
+    "DERIVEKEY",
+    "MACGENERATE",
+    "MACVERIFY",
+    "APPMANAGEABLE",
+    "EXPORT"
+  ]
+  enabled         = true
+  expiry_date     = "2025-02-02T17:04:05Z"
+  description     = "sobject description"
+  custom_metadata = {
+    key1 = "value1"
+  }
+  allowed_key_justifications_policy = [
+    "CUSTOMER_INITIATED_SUPPORT",
+    "CUSTOMER_INITIATED_ACCESS"
+  ]
+  allowed_missing_justifications = true
+  rotation_policy = {
+    interval_days = 20
+    effective_at = "20241130T183000Z"
+    deactivate_rotated_key = true
+    rotate_copied_keys = "all_external"
+  }
+}
+
+// Copy a security object in a group
+resource "dsm_sobject" "aes_sobject_example_copy" {
+  name            = "aes_sobject_example_copy"
+  group_id        = dsm_group.group.id
+  key = {
+    kid = dsm_sobject.aes_sobject_example.id
+  }
+  key_ops         = [
+    "ENCRYPT",
+    "DECRYPT",
+    "WRAPKEY",
+    "UNWRAPKEY",
+    "DERIVEKEY",
+    "MACGENERATE",
+    "MACVERIFY",
+    "APPMANAGEABLE",
+    "EXPORT"
+  ]
+  enabled         = true
+  expiry_date     = "2026-02-02T17:04:05Z"
+  description     = "sobject description copy"
+  custom_metadata = {
+    key1 = "value1"
+  }
+  rotation_policy = {
+    interval_days = 20
+    effective_at = "20241130T183000Z"
+    deactivate_rotated_key = true
+    rotate_copied_keys = "all_external"
+  }
+}
+
+```
+
+## Rotate a security object
+
+```terraform
+resource "dsm_sobject" "aes_sobject_example_rotate" {
+  name = "aes_sobject_example_rotate"
+  obj_type = "AES"
+  group_id = dsm_group.group.id
+  key_size = 256
+  key_ops = ["EXPORT", "ENCRYPT", "DECRYPT", "WRAPKEY", "UNWRAPKEY", "DERIVEKEY", "MACGENERATE", "MACVERIFY", "APPMANAGEABLE"]
+  rotate = "DSM"
+  // Name of the above security object
+  rotate_from = "aes_sobject_example"
+}
+```
+
+## Destruct a security object
+
+```terraform
+/*
+Destruction can be done while update only. 
+To destruct a security object, `destruct` parameter should be configured.
+And make enabled as false, this is to avoid the differences while updating other resources.
+*/
+resource "dsm_sobject" "aes_sobject_example" {
+  name            = "aes_sobject_example"
+  obj_type        = "AES"
+  group_id        = dsm_group.group.id
+  key_size        = 256
+  key_ops         = [
+    "ENCRYPT",
+    "DECRYPT",
+    "WRAPKEY",
+    "UNWRAPKEY",
+    "DERIVEKEY",
+    "MACGENERATE",
+    "MACVERIFY",
+    "APPMANAGEABLE",
+    "EXPORT"
+  ]
+  enabled         = false
+  expiry_date     = "2025-02-02T17:04:05Z"
+  description     = "aes sobject description"
+  custom_metadata = {
+    key1 = "value1"
+  }
+  allowed_key_justifications_policy = [
+    "CUSTOMER_INITIATED_SUPPORT",
+    "CUSTOMER_INITIATED_ACCESS"
+  ]
+  allowed_missing_justifications = true
+  rotation_policy = {
+    interval_days = 20
+    effective_at = "20241130T183000Z"
+    deactivate_rotated_key = true
+    rotate_copied_keys = "all_external"
+  }
+  destruct = "compromise" // other values: deactivate and destroy
```

Comment:
nit: it should be `"deactivate" or "destroy"` since it's either value not both (which some could interpret due to the word 'and').

Reply:
Changed